cmake/app/boilerplate.cmake

```diff
@@ -181,6 +181,13 @@ endif()
 assert(BOARD "BOARD not set")
 message(STATUS "Board: ${BOARD}")
 
+include(${ZEPHYR_BASE}/boards/deprecated.cmake)
+if(${BOARD}_DEPRECATED)
+  set(BOARD_DEPRECATED ${BOARD})
+  set(BOARD ${${BOARD}_DEPRECATED})
+  message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
```

Comment:
Let's move the message one line up so we don't have to mutate BOARD_DEPRECATED.

Reply:
Now that we are using `BOARD_DEPRECATED`, then I don't think we gain anything by moving those lines around.
https://github.com/zephyrproject-rtos/zephyr/blob/faa9b1fdd727f0c32d96a7991728c5fda1348c47/cmake/app/boilerplate.cmake#L161

Reply:
Just a small informative update.
The original reason I added `BOARD_DEPRECATED` was because I expected that other parts of Zephyr or downstream forks could find it useful to know if a deprecated board was specified and what the value were.